### ISSUE TITLE
Error with relative paths

### ISSUE THREAD

**Issue by nicoguaro** [nschloe/meshio#200]
I am trying to read a mesh that is in a folder above the current one using the following code

    import meshio

    fname = "../square.msh"
    points, cells, point_data, cell_data, field_data = \
         meshio.read(fname)

But I obtain the following error

    Traceback (most recent call last):
    
      File "<ipython-input-5-c5965f4f0bf1>", line 1, in <module>
        runfile('/home/nguarinz/Documents/meshio_test.py', wdir='/home/nguarinz/Documents')
    
      File "/home/nguarinz/anaconda2/lib/python2.7/site-packages/spyder/utils/site/sitecustomize.py", line 705, in runfile
        execfile(filename, namespace)
    
      File "/home/nguarinz/anaconda2/lib/python2.7/site-packages/spyder/utils/site/sitecustomize.py", line 94, in execfile
        builtins.execfile(filename, *where)
    
      File "/home/nguarinz/Documents/meshio_test.py", line 6, in <module>
        meshio.read(fname)
    
      File "/home/nguarinz/anaconda2/lib/python2.7/site-packages/meshio/helpers.py", line 99, in read
        file_format = _extension_to_filetype[extension]
    
    KeyError: '../square.msh'

Checking lin 99 of [helpers.py](https://github.com/nschloe/meshio/blob/master/meshio/helpers.py) (98 in the current branch), it seems to be because it takes the extension as all the characters after the first point. I suppose that something like this

    extension = filename[filename.rfind("."):]

might work.

I am using meshio 1.11.2 with Python 2.7.13 (Anaconda 64 bits) in Linux Mint 18.2.

**Comment by gdmcbain** [comment]
I was going to propose the more direct [`os.path.splitext`](https://docs.python.org/3/library/os.path.html?highlight=splitext#os.path.splitext), but saw that it had been commented out in the corresponding part of `helpers.write` in d2411cce1f4b6c08a1e3131187b49b20025243d5:

```python
        # _, extension = os.path.splitext(filename)
        extension = '.' + filename.split(os.extsep, 1)[-1]
```

I think that that would have worked in your case, i.e. `os.path.splitext('../square.msh')[-1] == '.msh'`.

**Comment by gdmcbain** [comment]
Ah, I see from the change to test/test_helpers.py in the same commit that it was to allow double-extensions like '.post.gz' for filename 'test.post.gz'.  In which case [`os.path.basename`](https://docs.python.org/3/library/os.path.html?highlight=basename#os.path.basename) should fix the present issue.

```python
        extension = os.extsep + os.path.basename(filename).split(os.extsep, 1)[-1]
```

**Comment by nschloe** [comment]
Fixed by #201.
